### PR TITLE
Add server without database

### PR DESCRIPTION
**Pull request by smesseim**
The server can now be started without using the database. All bubbles
are calculated on the fly.

To start the server without database, pass the arguments `8081 TB10
nodatabase`. Otherwise pass `8081 TB10 database`.

### CODE REVIEW

**Comment by SamuelSital** [comment]
LGTM

File: src/main/java/io/github/programminglife2016/pl1_2016/database/FetchDatabase.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ private JSONArray fetchLinks(int threshold) throws SQLException {
      * @return the collection of specimen
      * @throws SQLException thrown if SQL connection or query is not valid
      */
-//    @SuppressWarnings("checkstyle:methodlength")
```
Comment (RaviShivam) [inline src/main/java/io/github/programminglife2016/pl1_2016/database/FetchDatabase.java]: This method is not that large to throw a checkstyle error. I don't think that it is necessary to suppress this.

File: src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/NodeMap.java [hunk 2]
```diff
@@ -24,19 +24,23 @@ public final String toJson() {
...
             Node node = entry.getValue();
             Node start = retrieveSegment(node, true);
             Node end = retrieveSegment(node, false);
-            int x = (start.getX() + end.getX())/2;
-            int y = (start.getY() + end.getY())/2;
+            int x = (start.getX() + end.getX()) / 2;
+            int y = (start.getY() + end.getY()) / 2;
             this.get(entry.getKey()).setXY(x, y);
         }
     }
```
Comment (RaviShivam) [inline src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/NodeMap.java]: Is it really necessary to expand the if statement? The one-liner seemed to work just as well

File: src/main/java/io/github/programminglife2016/pl1_2016/server/api/queries/GetThresholdedBubblesApiQuery.java [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
 import io.github.programminglife2016.pl1_2016.server.api.actions.ApiAction;
 
 /**
  * Listens to /api/nodes/[threshold] and return the data of segment [threshold].
  */
 public class GetThresholdedBubblesApiQuery implements ApiQuery {
-    private FetchDatabase fdb;
+    private BubbleDispatcher bubbleDispatcher;
 
     /**
```
Comment (RaviShivam) [inline src/main/java/io/github/programminglife2016/pl1_2016/server/api/queries/GetThresholdedBubblesApiQuery.java]: There is no parameter

File: src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/NodeMap.java (hunk 2 above)
Comment (SamuelSital) [inline src/main/java/io/github/programminglife2016/pl1_2016/parser/nodes/NodeMap.java]: Checkstyle will throw an error for inline ternary if-statements

File: src/main/java/io/github/programminglife2016/pl1_2016/database/FetchDatabase.java (hunk 1 above)
Comment (SamuelSital) [inline src/main/java/io/github/programminglife2016/pl1_2016/database/FetchDatabase.java]: Removed the annotation.

**Comment by RaviShivam** [comment]
LGTM

**Comment by smesseim** [comment]
Can't really say I'm a fan of setter chaining, but you can leave it in.

Can you remove BubbleMain altogether? Not useful any more.

**Pull request merged**